File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    for a in arr:
        if not answer or answer[-1] != a:
            answer.append(a)            
    # [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    print('Hello Python')
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> answer = new ArrayList<>();
        int preNum = -1;
        
        for(int a:arr){
            if(a!=preNum){
                answer.add(a);
                preNum = a;
            }
        }
        
        return answer.stream().mapToInt(i->i).toArray();
    }
}